src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py

```diff
@@ -50,20 +52,27 @@ def validate_registry_name_create(namespace):
 def validate_storage_account_name(namespace):
     client = storage_client_factory().storage_accounts
 
-    if namespace.storage_account_name is None:
+    if namespace.storage_account_name:
+        storage_account_name = namespace.storage_account_name
+        if arm_get_storage_account_by_name(storage_account_name) is None:
+            raise CLIError('The storage account {} does not exist'.format(storage_account_name))
```

Comment:
We are not asking the user to provide the resource group for a storage account. We are querying ARM for a storage account and ARM will return None if the storage account doesn't exist. We cannot even make the RP call to storage account client because we don't have resource group information. So we are handling it here. There is no client side logic but a simple ARM call to find out if the storage account exists. Besides, we can have a more friendly error/warning message to point the user to the direction of creating a new storage account.